### PR TITLE
Generate Geo-aware scores for all zoom levels

### PR DESCRIPTION
**Pull request by esfoobar-usds**
Work for #332

### CODE REVIEW

**Review by NatHillardUSDS — APPROVED** [review]
Awesome - a great change! Looking forward to this.

File: data/data-pipeline/etl/runner.py [hunk 1]
```diff
@@ -107,6 +108,23 @@ def score_generate() -> None:
...
+    Returns:
+        None
+    """
+
+    # Score Geo
+    score_geo = GeoScoreETL()
+    score_geo.extract()
+    score_geo.transform()
+    score_geo.load()
+
```
Comment (ginabeena) [inline data/data-pipeline/etl/runner.py:126]: E - T - L ...simple enough 😉

File: data/data-pipeline/etl/score/etl_score_geo.py [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        self.geojson_score_usa_high: gpd.GeoDataFrame
+        self.geojson_score_usa_low: gpd.GeoDataFrame
+
+    def extract(self) -> None:
+        logger.info(f"Reading US GeoJSON (~6 minutes)")
+        self.geojson_usa_df = gpd.read_file(
+            self.CENSUS_USA_GEOJSON,
+            dtype={"GEOID10": "string"},
+            usecols=["GEOID10", "geometry"],
+            low_memory=False,
```
Comment (ginabeena) [inline data/data-pipeline/etl/score/etl_score_geo.py:44]: what happens if you pass true 👀

File: data/data-pipeline/etl/score/etl_score_geo.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        )
+
+    def transform(self) -> None:
+        logger.info(f"Pruning Census GeoJSON")
+        fields = ["GEOID10", "geometry"]
+        self.geojson_usa_df = self.geojson_usa_df[fields]
+
+        logger.info(f"Merging and compressing score CSV with USA GeoJSON")
+        self.geojson_score_usa_high = self.score_usa_df.merge(
+            self.geojson_usa_df, on="GEOID10", how="left"
```
Comment (ginabeena) [inline data/data-pipeline/etl/score/etl_score_geo.py:62]: might be nice to comment what these literals mean

File: data/data-pipeline/etl/score/etl_score_geo.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        )
+
+        self.geojson_score_usa_high = gpd.GeoDataFrame(
+            self.geojson_score_usa_high, crs="EPSG:4326"
```
Comment (ginabeena) [inline data/data-pipeline/etl/score/etl_score_geo.py:66]: same sentiment for `EPSG:4326`

File: data/data-pipeline/etl/score/etl_score_geo.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+        return state_tracts
+
+    def _create_buckets_from_tracts(
+        self, state_tracts: gpd.GeoDataFrame, num_buckets: int
+    ) -> gpd.GeoDataFrame:
+        # assign tracts to buckets by D_SCORE
+        state_tracts.sort_values(self.TARGET_SCORE_RENAME_TO, inplace=True)
+        SCORE_bucket = []
+        bucket_size = math.ceil(
+            len(state_tracts.index) / self.NUMBER_OF_BUCKETS
```
Comment (ginabeena) [inline data/data-pipeline/etl/score/etl_score_geo.py:122]: what if number of buckets is 0

**Review by ginabeena — APPROVED** [review]
Looks good to me @esfoobar-usds, nice work :shipit:

File: data/data-pipeline/etl/score/etl_score_geo.py (hunk 2 above)
Comment (esfoobar-usds) [reply data/data-pipeline/etl/score/etl_score_geo.py:44]: It will load in batches, thus much more slowly.

**Pull request merged**